src/AutoRest.CSharp/Mgmt/AutoRest/MgmtOutputLibrary.cs

```diff
@@ -225,25 +226,19 @@ private static void IncrementCount(Dictionary<Schema, int> usageCounts, Schema s
         // Initialize ResourceData, Models and resource manager common types
         private Dictionary<Schema, TypeProvider> InitializeModels()
         {
-            var resourceModels = new Dictionary<Schema, TypeProvider>();
-
             // first, construct resource data models
             foreach (var schema in _allSchemas)
             {
                 var model = ResourceDataSchemaNameToOperationSets.ContainsKey(schema.Name) ? BuildResourceModel(schema) : BuildModel(schema);
-                resourceModels.Add(schema, model);
-                _nameToTypeProvider.Add(schema.Name, model); // TODO: ADO #5829 create new dictionary that allows look-up with multiple key types to eliminate duplicate dictionaries
+                _nameToTypeProvider.Add(schema, model);
```

Comment:
Maybe we should name it after `_schemaOrNameToResourceModels`? because:
1. the key can be either schema or string
2. `TypeProvider` are actually resource models